src/hooks/core/chat/useChatUtilsHook.ts

```diff
@@ -135,12 +135,15 @@ function useChatUtilsHook(): UseChatUtilsHook {
     return null;
   };
 
-  const goBackFromChatScreen = () => {
-    navigation.goBack();
-    setChat({
-      ...chat,
-      selectedChannel: null
-    });
+  const goBackFromChatScreen = async () => {
+    await Keyboard.dismiss();
+    setTimeout(() => {
+      navigation.goBack();
+      setChat({
+        ...chat,
+        selectedChannel: null
+      });
+    }, 350);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The use of `setTimeout` here could potentially lead to a race condition if the user navigates away from the chat screen before the timeout completes. This could result in an error or unexpected behavior. Consider using a Promise-based approach to ensure that the keyboard has been dismissed before proceeding with navigation and state updates.
```diff
-    await Keyboard.dismiss();
-    setTimeout(() => {
-      navigation.goBack();
-      setChat({
-        ...chat,
-        selectedChannel: null
-      });
-    }, 350);
+    Keyboard.dismiss().then(() => {
+      navigation.goBack();
+      setChat({
+        ...chat,
+        selectedChannel: null
+      });
+    });
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->